pkg/daemon/drain_test.go

```diff
@@ -359,9 +361,15 @@ location = "example.com/repo/test-img"
 				t.Errorf("parsing new Ignition config failed: %v", err)
 			}
 			diffFileSet := ctrlcommon.CalculateConfigFileDiffs(&oldIgnConfig, &newIgnConfig)
-			drain, err := isDrainRequired(test.actions, diffFileSet, oldIgnConfig, newIgnConfig)
-			if !reflect.DeepEqual(test.expectedAction, drain) {
-				t.Errorf("Failed determining drain behavior: expected: %v but result is: %v. Error: %v", test.expectedAction, drain, err)
+			readOldFile := func(path string) ([]byte, error) {
```

Comment:
I noticed you duplicated these in a few places. A technique to reduce duplication is to have a function that returns a function (I know it sounds strange):

```golang
func getIgnitionFileDataReadFunc(ignConfig *ign3types.IgnConfig, path string) ReadFileFunc {
  return func(path string) ([]byte, error) {
    return ctrlcommon.GetIgnitionFileDataByPath(ignConfig, path)
  }
}
```

Then your calls become something like:

```golang
readOldFile := getIgnitionFileDataReadFunc(&oldIgnConfig, path)
readNewFile := getIgnitionFileDataReadFunc(&newIgnConfig, path)
```

Reply:
Ooo I like that more! Changed